src/NLog/Targets/FileArchiveModes/FileArchiveModeDate.cs

```diff
@@ -86,6 +86,7 @@ public override DateAndSequenceArchive GenerateArchiveFileName(string archiveFil
             FileNameTemplate archiveFileNameTemplate = GenerateFileNameTemplate(archiveFilePath);
             string dirName = Path.GetDirectoryName(archiveFilePath);
             archiveFilePath = Path.Combine(dirName, archiveFileNameTemplate.ReplacePattern("*").Replace("*", archiveDate.ToString(_archiveDateFormat)));
+            archiveFilePath = Path.GetFullPath(archiveFilePath);    // Rebuild to fix non-standard path-format
```

Comment:
I'm afraid this could be slow. AFAIK GetFullPath is a really slow method.

That's one of the reasons FilePathLayout was build. See https://github.com/NLog/NLog/blob/0bfdad5213e0dccc6fc3ef2b192d5446411953b3/src/NLog/Internal/Files/FilePathLayout.cs#L233-L234

Reply:
It only happens on archive operations. Archive-operations are by default slow. As correctly observed by #3608 then `Path.GetFullPath` happens for some archive-modes, but not all. This PR aligns them all.

Reply:
The method you are referring to is called for every LogEvent. These methods are callled for every archive-operation, which is a different number.

Reply:
OK 👍 thanks for checking/explaining!